pkg/repository/repository_test.go

```diff
@@ -23,12 +25,18 @@ import (
 )
 
 type fakeRepositoryGetter struct {
-	repository *fakeRepository
-	err        error
+	repository []*fakeRepository
+	err        []error
+	callCount  int
 }
 
-func (frg fakeRepositoryGetter) get(_ string) (gitRepositoryInterface, error) {
-	return frg.repository, frg.err
+func (frg *fakeRepositoryGetter) get(_ string) (gitRepositoryInterface, error) {
```

Comment:
This is required because we call this function several times and we need to mock a sequennce of responses/errors instead of a mock value all the time